### ISSUE TITLE
salt-ssh: multiple targets fails after upgrade to 2017.7

### ISSUE THREAD

**Issue by doublez13** [saltstack/salt#43643]
### Description of Issue/Question
After upgrading from 2016.11 to 2017.7, salt-ssh doesn't work when targeting multiple minions.

salt-ssh host1,host2,host3 cmd.run 'echo $HOSTNAME'
will just call one of the hosts 3 times

I have 75 hosts in my roster.
salt-ssh '*' cmd.run 'echo $HOSTNAME'
makes the same host respond 75 times.

### Steps to Reproduce Issue
salt-ssh host1,host2,host3 cmd.run 'echo $HOSTNAME'

### Versions Report
Salt Version:
           Salt: 2017.7.1
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.9.6
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.7
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.10 (default, Dec  2 2015, 11:55:42)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.2
            ZMQ: 4.1.6
 
System Versions:
           dist: centos 7.4.1708 Core
         locale: UTF-8
        machine: x86_64
        release: 3.10.0-693.2.2.el7.x86_64
         system: Linux

**Comment by Ch3LL** [comment]
I cannot replicate this:

```
➜  salt git:(0931281ebd) sudo salt-ssh '*' cmd.run 'echo $HOSTNAME'
ch3ll1:
    ip-10-156-122-152
ch3ll2:
    ip-10-111-201-166
```

How did you upgrade/install salt?

Also if you add `-ldebug` to the output does it provide more information about this issue?

**Comment by doublez13** [comment]
I did a 'pip install --upgrade salt' from 2016.11.
I can give salt-ssh a list of hosts in the roster, match based on partial hostnames, or any other method to specify multiple hosts, and it just does the same host each time.

**Comment by doublez13** [comment]
salt-ssh -L host1,host2,host3 cmd.run 'echo $HOSTNAME' -ldebug

This outputs a bunch of lines, but the three that stand out to me are...

Terminal Command: /bin/sh -c ssh host1 -t -t -o KbdInteractiveAuthentication=no -o PasswordAuthentication=no -o GSSAPIAuthentication=no -o ConnectTimeout=65 -o Port=22 -o IdentityFile=/root/.ssh/id_rsa -o User=root  mkdir -p

This line appears three times, but every time it SSHes to host1. Seems like it should be three lines with host1, host2, and host3 respectively

**Comment by doublez13** [comment]
Just did a completely fresh install of salt 2017.7.1 on a fresh install of python 2.7.14
Copied over master.conf and my roster file.

Same results!!

`[root@salt python]# salt-ssh host[1-3] cmd.run 'echo $HOSTNAME'
host1:
    host2
host3:
    host2
host2:
    host2
`

**Comment by Ch3LL** [comment]
from that last command output i'm not seeing it echo teh hostname? I would assume that it would echo the host1's hostname?

Can you run salt-ssh in debug mode and see where it states: `Matched minions:` and see if its listing out just host1?

Also if you do a package install does it make a difference?

**Comment by rallytime** [comment]
@doublez13 This should be fixed with #43889. Can you confirm?
